test/api_bridge_test.rb

```diff
@@ -0,0 +1,165 @@
+    opts = @api.calc_start({"page" => "-1", "num" => "10"})
+    assert_equal 0, opts["start"]
+  end
+
+  test "create_item_url" do
+    url = @api.create_item_url("cat.let1728")
+    assert_equal "https://test.unl.edu/v1/item/cat.let1728", url
+  end
+
+  test "create_items_url" do
+    # just a facet
+    url = @api.create_items_url({ "facet" => ["date.year", "source"] })
+    assert_equal "https://test.unl.edu/v1/items?facet[]=date.year&facet[]=source", url
+    # a filter that needs some encoding
+    url = @api.create_items_url({ "f" => ["creator.name|\"Yosemite Sam\": Worst Name Ever"]})
+    assert_equal "https://test.unl.edu/v1/items?f[]=creator.name%7C%22Yosemite+Sam%22%3A+Worst+Name+Ever", url
+    # a bunch of things
+    url = @api.create_items_url({ "q" => "text:water", "hl_num" => "3", "f" => ["category|Writings"]})
+    assert_equal "https://test.unl.edu/v1/items?q=text%3Awater&hl_num=3&f[]=category%7CWritings", url
+  end
+
+  test "encode_param" do
+    # no particularly unusual characters
+    param = @api.encode_param("f[]=source|UNL Special Collections")
+    assert_equal "f[]=source%7CUNL+Special+Collections", param
+    # ampersand!
+    param = @api.encode_param("f[]=repository|Archives & Special Collections")
+    assert_equal "f[]=repository%7CArchives+%26+Special+Collections", param
+    # numbers
+    param = @api.encode_param("facet[]=date.year|1867")
+    assert_equal "facet[]=date.year%7C1867", param
+  end
+
+  # get_item_by_id
+
+  test "is_url?" do
+    # definitely not
+    assert_not @api.is_url?("some string")
+    # sort of but not quite!
+    assert_not @api.is_url?("test.unl.edu")
+    # http
+    assert @api.is_url?("http://test.unl.edu")
+    # https
+    assert @api.is_url?("https://test.unl.edu")
+    # port and path
+    assert @api.is_url?("http://localhost:3000/api/v1/items")
+  end
+
+  test "params_to_hash" do
+    # make sure this hash is not altered and returns only as string keys
+    original = { "a" => 1, b: 2 }
+    cloned = @api.params_to_hash(original)
+    assert_includes cloned, "b"
+    assert_not_includes cloned, :b
+    cloned.delete("a")
+    assert_includes original, "a"
+    # TODO ActionController::Parameters object next!
+    original = ActionController::Parameters.new({ "a" => 1, b: 2 })
+    cloned = @api.params_to_hash(original)
+    assert_includes cloned, "b"
+    assert_not_includes cloned, :b
+    cloned.delete("a")
+    assert_includes original, "a"
+  end
+
+  test "prepare_options" do
+    # make sure rows does not override num when also sent in
+    # (@api created with num)
+    opts = @api.prepare_options({ "rows" => 20 })
+    assert_not_includes opts, "rows"
+    assert_equal 5, opts["num"]
+    # TODO figure out if we want rows to override num even though
+    # there is a default num set...will need to figure something out
+    # to determine if num was the default or passed in....??
+    # api = ApiBridge::Query.new("https://test.unl.edu/v1", [], {})
```

Comment:
Note: Resolved by decision to always overwrite in newer code but I'll leave the note here fyi

Could set a boolean attribute on `api` when initializing to determine whether num was set with the default or not

Reply:
that's true.  since we only have about 3 sites using this it's probably okay, but good idea if we do need to change something fairly major in the future!